packages/hdwallet-ledger-webusb/src/adapter.ts

```diff
@@ -45,56 +51,60 @@ export class WebUSBLedgerAdapter {
   private async handleDisconnectWebUSBLedger(e: USBConnectionEvent): Promise<void> {
     if (e.device.vendorId !== VENDOR_ID) return;
 
+    this.keyring.emit(
+      [e.device.manufacturerName ?? "", e.device.productName ?? "", core.Events.DISCONNECT],
+      e.device.serialNumber
+    );
+
     const ts = Date.now();
     this.currentEventTimestamp = ts;
 
     // timeout gives time to detect if it is an app navigation based disconnect/connect event
     // discard disconnect event if it is not the most recent event received
-    setTimeout(async () => {
-      if (ts !== this.currentEventTimestamp) return;
-
-      try {
-        if (e.device.serialNumber) await this.keyring.remove(e.device.serialNumber);
-      } catch (error) {
-        console.error(error);
-      } finally {
-        this.keyring.emit(
-          [e.device.manufacturerName ?? "", e.device.productName ?? "", core.Events.DISCONNECT],
-          e.device.serialNumber
-        );
-      }
-    }, APP_NAVIGATION_DELAY);
+    // TODO(gomes): maybe uncomment me maybe not
+    // setTimeout(async () => {
+    // if (ts !== this.currentEventTimestamp) return;
+    //
+    // try {
+    // if (e.device.serialNumber) await this.keyring.remove(e.device.serialNumber);
+    // } catch (error) {
+    // console.error(error);
+    // }
+    // }, APP_NAVIGATION_DELAY);
   }
 
   public get(device: USBDevice): ledger.LedgerHDWallet {
     return core.mustBeDefined(this.keyring.get<ledger.LedgerHDWallet>(device.serialNumber));
   }
 
   // without unique device identifiers, we should only ever have one ledger device on the keyring at a time
-  public async initialize(usbDevice?: USBDevice): Promise<number> {
-    const device = usbDevice ?? (await getFirstLedgerDevice());
+  public async initialize(ledgerTransport?: TransportWebUSB): Promise<number> {
```

Comment:
Initialize with a `transport` vs. a `deviceId` meaning we don't run the connect logic twice